backend/modules/database/db_functions.bal

```diff
@@ -14,14 +14,42 @@
 // specific language governing permissions and limitations
 // under the License. 
 import ballerina/sql;
+import ballerina/log;
 
 # Fetch all apps visible to the given roles.
 #
 # + return - App[] or an error? on failure
 public isolated function fetchApps() returns App[]|error {
-    stream<App, error?> result = databaseClient->query(fetchAppsQuery());
-    return from App app in result
-        select app;
+    App[] apps = [];
+    stream<AppStr, error?> result = databaseClient->query(fetchAppsQuery());
```

Comment:
if this is retruning one app, we don not need to process a stream. lest have a quick review.